lxd/storage/backend_lxd.go

```diff
@@ -1645,9 +1667,28 @@ func (b *lxdBackend) CreateInstanceFromImage(inst instance.Instance, fingerprint
 	revert := revert.New()
 	defer revert.Fail()
 
+	volumeConfig := make(map[string]string)
+	err = b.applyInstanceRootDiskInitialValues(inst, volumeConfig)
+	if err != nil {
+		return err
+	}
+
+	useOptimizedImage := b.driver.Info().OptimizedImages
```

Comment:
Could we push this part and the if statement on line 1682 into the `shouldUseOptimizedImage` function, as its all part of the logic surrounding the decision to use optimized image or not, then the call here would become just:

```go
volumeConfig := make(map[string]string)
err = b.applyInstanceRootDiskInitialValues(inst, volumeConfig)
if err != nil {
    return err
}

useOptimizedImage, err = b.shouldUseOptimizedImage(fingerprint, contentType, volumeConfig, op)
if err != nil {
    return err
}
```